backend/estimate-service/src/main/java/org/egov/works/validator/EstimateServiceValidator.java

```diff
@@ -236,4 +240,64 @@ public void validateSearchEstimate(RequestInfoWrapper requestInfoWrapper, Estima
             throw new CustomException("TENANT_ID", "Tenant is mandatory");
         }
     }
+
+    public void validateUpdateEstimate(EstimateRequest request) {
+        Map<String, String> errorMap = new HashMap<>();
+        Estimate estimate = request.getEstimate();
+        RequestInfo requestInfo = request.getRequestInfo();
+        EstimateRequestWorkflow workflow = request.getWorkflow();
+
+        validateRequestInfo(requestInfo, errorMap);
+        validateEstimate(estimate, errorMap);
+        validateWorkFlow(workflow, errorMap);
+
+        UUID id = estimate.getId();
+        if (id == null) {
+            errorMap.put("ESTIMATE_ID", "Estimate id is mandatory");
+        } else {
+            List<String> ids = new ArrayList<>();
+            ids.add(id.toString());
+            EstimateSearchCriteria searchCriteria = EstimateSearchCriteria.builder().ids(ids).tenantId(estimate.getTenantId()).build();
+            List<Estimate> estimateList = estimateRepository.getEstimate(searchCriteria);
+            if (CollectionUtils.isEmpty(estimateList)) {
+                throw new CustomException("INVALID_ESTIMATE_MODIFY", "The record that you are trying to update does not exists in the system");
+            }
+        }
+
+        validateRolesForUpdateEstimate(requestInfo, errorMap);
+
+        String rootTenantId = estimate.getTenantId();
+        //split the tenantId
+        rootTenantId = rootTenantId.split("\\.")[0];
+
+        Object mdmsData = mdmsUtils.mDMSCall(request, rootTenantId);
+
+        validateMDMSData(estimate, mdmsData, errorMap);
+        validateLocation(estimate, requestInfo, errorMap);
+
+        if (!errorMap.isEmpty())
+            throw new CustomException(errorMap);
+
+    }
+
+    private void validateRolesForUpdateEstimate(RequestInfo requestInfo, Map<String, String> errorMap) {
+        User userInfo = requestInfo.getUserInfo();
+        if (userInfo.getRoles() == null || userInfo.getRoles().isEmpty()) {
+            errorMap.put("USER_ROLE", "User's role is missing");
+        } else {
+            List<org.egov.common.contract.request.Role> roles = userInfo.getRoles();
+            List<String> updateRoles = Arrays.asList(UPDATE_ROLES.split(","));
+
+            boolean rolePresent = roles.stream().anyMatch(role -> {
+                return updateRoles.contains(role.getCode());
+            });
+
+            List<String> roleCodes = roles.stream().map(role -> {
+                return role.getCode();
+            }).collect(Collectors.toList());
+            if (!rolePresent) {
+                errorMap.put("UNAUTHORIZED_ROLE", "User role(s) : " + roleCodes + " are not authorized to update estimate.");
```

Comment:
Most probably, this will not serve our purpose. 
The requirement is that - roles like tech sanctioner can call /_update. (And this is handled at zuul.) 
These roles will not actually update the estimate but only take action on it. By taking an action, the estimateStatus will automatically get updated. 
Only the EST_CREATOR role will take action as well as update the details of the estimate.

Reply:
Will discuss once..

Reply:
done here ..  https://github.com/egovernments/DIGIT-Works/pull/48/commits/211ce35a6f95226d3a85fe2ca993254db1086109